master/internal/api_user.go

```diff
@@ -427,19 +431,23 @@ func (a *apiServer) PatchUser(
 
 	var ug *model.AgentUserGroup
 	if aug := req.User.AgentUserGroup; aug != nil {
-		if err = validateProtoAgentUserGroup(aug); err != nil {
-			return nil, err
-		}
-		ug = &model.AgentUserGroup{
-			UID:   int(*req.User.AgentUserGroup.AgentUid),
-			GID:   int(*req.User.AgentUserGroup.AgentGid),
-			User:  *req.User.AgentUserGroup.AgentUser,
-			Group: *req.User.AgentUserGroup.AgentGroup,
-		}
+		if checkEmptyProtoAgentUserGroup(req.User.AgentUserGroup) {
+			ug = &model.AgentUserGroup{}
```

Comment:
When updating user, `AgentUserGroup: nil` means do not change agent user group, and `AgentUserGroup: {}` means remove agent user group if there is any